File: Medium/happiness_number.py
# https://edabit.com/challenge/8qD23E6XRMaWhyJ5z
def happiness_number(s):
    unhappy =0
    happy=0
    happy+=s.count(":)")
    happy+=s.count("(:")
    unhappy-=s.count(":(")
    unhappy-=s.count("):")
    return unhappy+happy


print(happiness_number(':):('), -1)
print(happiness_number('(:)'), 2)
print(happiness_number('::::'), 0)
print(happiness_number(':)::(()::'), -2)
print(happiness_number('))):'), -1)
print(happiness_number(':):)('), 1)
print(happiness_number(':(:(:()):'), -2)
print(happiness_number('()((:())):'), -1)
print(happiness_number(':(:)'), 1)
print(happiness_number('(:):(:)(('), 2)
print(happiness_number(':(:(:'), 0)
print(happiness_number(')())(()'), 0)
print(happiness_number(':()):'), -2)
print(happiness_number('():(('), -2)
print(happiness_number('::::(():('), -3)
print(happiness_number(')(():((::'), -1)
print(happiness_number('(::):('), 0)

File: javascript/Hard/happinessNumber.js
// https://edabit.com/challenge/eCmEgrKNtnfhpssQp

function happinessNumber(s) {
    const happy=s.match(/:\)/g)?s.match(/:\)/g).length:0
    const happy2=s.match(/\(:/g)?s.match(/\(:/g).length:0
    const sad = s.match(/:\(/g)?s.match(/:\(/g).length:0
    const sad2 = s.match(/\):/g)?s.match(/\):/g).length:0
    // !=null?s.match(/:\)|\(:/g).length:0
    // !=null?s.match(/:\(|\):/g).length:0 
	// return [s.match(/:\)|\(:/g),s.match(/:\(|\):/g)]
    return (happy+happy2)-(sad+sad2)
    // |\(:
}

console.log(happinessNumber('::::(():('), -3)
console.log(happinessNumber(':):('), -1)
console.log(happinessNumber('(:)'), 2)
console.log(happinessNumber('::::'), 0)
console.log(happinessNumber(':)::(()::'), -2)